sample_consensus/include/pcl/sample_consensus/sac_model_plane.h

```diff
@@ -257,13 +257,45 @@ namespace pcl
       using SampleConsensusModel<PointT>::sample_size_;
       using SampleConsensusModel<PointT>::model_size_;
 
+      /** This implementation uses no SIMD instructions. It is not intended for normal use.
+        * See countWithinDistance which automatically uses the fastest implementation.
+        */
+      std::size_t
+      countWithinDistanceStandard (const Eigen::VectorXf &model_coefficients,
+                                   const double threshold,
+                                   std::size_t i = 0) const;
+
+      /** This implementation uses SSE, SSE2, and SSE4.1 instructions. It is not intended for normal use.
+        * See countWithinDistance which automatically uses the fastest implementation.
+        */
+      std::size_t
+      countWithinDistanceSSE (const Eigen::VectorXf &model_coefficients,
+                              const double threshold,
+                              std::size_t i = 0) const;
```

Comment:
Maybe add an end option via an overload? That'll allow generic (seems to work) OpenMP implementation that can internally use the AVX and SSE functions.

[godbolt link](https://godbolt.org/z/cKLTvv)

Do note that I used tags, but you can skip that.

Reply:
That would be possible, but TBH I would leave that to the person who creates that OpenMP implementation, if that ever happens. Right now it would be just unused code.

Reply:
> Right now it would be just unused code.

Sure